client/src/components/common/boost/BoostConfigurator.tsx

```diff
@@ -23,17 +25,49 @@ export default function BoostConfigurator({
   onDeleteConfig,
   onUploadConfig,
 }: BoostConfiguratorProps) {
+  const fileInput = createRef<HTMLInputElement>();
+  const [configType, setConfigType] = useState<BoostConfigType>('all');
+
+  // Function to click the hidden file input button (this is a work around to avoid the ugly
+  // UI of the default input file button)
+  const handleClick = (type: BoostConfigType) => {
+    if (fileInput.current) {
+      setConfigType(type);
+      fileInput.current.click();
+    }
+  };
+
+  const handleFileUpload = (event: React.ChangeEvent<HTMLInputElement>) => {
+    // This method gets called even if the user uploads a file and the second time they cancel
+    // to upload, hence the need to check that the files attribute is not an array of 0 length
+    if (event.target.files != null && event.target.files.length !== 0) {
+      onUploadConfig(configType, event.target.files);
+    }
+  };
+
   return (
-    <Card>
+    <Card style={{ marginTop: '2.5rem' }}>
       <Card.Body>
         <Card.Title>Configuration</Card.Title>
-        <Button
-          variant="outline-primary"
-          className="mb-3"
-          onClick={onUploadConfig}
-        >
-          Upload config
-        </Button>
+        <>
+          <input
+            ref={fileInput}
+            onChange={handleFileUpload}
+            type="file"
+            style={{ display: 'none' }}
+            accept=".json"
+            multiple={false}
+          />
+          <Button
+            variant="outline-primary"
+            className="mb-3"
+            onClick={() => {
+              handleClick('all');
+            }}
```

Comment:
Not at all an issue, but in case you didn't know, you can remove the braces around one liners like these: `onClick={() => handleClick('all')}`

(if we were using a returned value, we also get to omit `return` with this syntax. Neat!)